File: JavaNgArch/src/javangarch/domain/domainmodel/gn/Dept.java
package javangarch.domain.domainmodel.gn;

import java.util.List;

import javax.persistence.JoinColumn;
import javax.persistence.JoinTable;
import javax.persistence.OneToMany;

import lombok.Getter;
import lombok.Setter;
import projectbase.domain.BaseDomainObject;
import projectbase.domain.RefText;
import projectbase.domain.customcollection.IDomainList;
import projectbase.sharparch.domain.domainmodel.DomainSignature;
@Getter
@Setter
public class Dept extends BaseDomainObject {
	/**
	 * 
	 */
	private static final long serialVersionUID = 1L;
	public static final String DefaultSort = "this.code asc";

	@DomainSignature
	private Corp corp;
	@DomainSignature
	private String code;

	@RefText
	private String name;

	private IDomainList<User> users;
	public List<User> getUsers() {
		return users;
	}

	private List<Func> funcs;
	@OneToMany
	@JoinTable(	name="gn_deptfunc",joinColumns=@JoinColumn(name="DeptID"),inverseJoinColumns=@JoinColumn(name="FuncID")	)
	public List<Func> getFuncs() {
		return funcs;
	}

}

File: NetNgArch/src/NetNgArch.Domain/DomainModel/GN/Dept.cs
using System;
using System.Collections.Generic;
using ProjectBase.Domain;
using ProjectBase.Utils;
using SharpArch.Domain.DomainModel;

namespace NetNgArch.Domain.DomainModel.GN
{
    public class Dept : BaseDomainObject
    {
        public static readonly string DefaultSortString = "Code";
        public static readonly SortStruc<Dept>[] DefaultSort = new SortStruc<Dept>[]
                                                                    {
                                                                        new SortStruc<Dept>{
                                                                            OrderByExpression=o => o.Code,
                                                                            OrderByDirection = OrderByDirectionEnum.Asc
                                                                        }
                                                               };
        #region "persistent properties"

		[DomainSignature]
		public virtual Corp Corp { get; set; }
		[DomainSignature]
		public virtual string Code { get; set; }
        [RefText]
		public virtual string Name { get; set; }

        public virtual IDomainList<User> Users { get; set; }
        public virtual IList<Func> Funcs { get; set; } 
     
        #endregion


        #region "methods"

        #endregion

        #region "Enums"

        #endregion
    }

}

